### PR TITLE
Save config with command line arguments

### LINKED ISSUE #9: Config file in output should reflect actual arguments used
Currently, during training, saber creates an output folder and saves the `config.ini` file used to specify model/training details to this output folder. However, if saber was called in the command line and command line args were passed in this call, this is not captured in that saved config file.

- [x] The `config.ini` file saved to the output folder during training should reflect any arguments passed at the command line.

### PR DESCRIPTION
**Pull request by JohnGiorgi**
When experimenting (i.e. training many models and observing the effect of different model settings, hyper-parameters, training strategies, etc) it can quickly become hard to keep track of each individual change. To this end, saber takes the `config.ini` file used when the model is initialized and saves it to the output folder of the training session. Previously however, this did not account for any arguments supplied at the command line. This pull request:

- Adds a new method, `Config.save()` which saves a `config.ini` file to the output folder of a training session. This `config.ini` contains the training settings / hyperparameters.
- Of much lesser importance, no longer normalizes empty file-path arguments in the 'config.ini' file to None

Left to do:

- [x] Write a unit test to for `Config.save()`.
- [x] Pin `Keras==2.2.0` to stop build errors.
- [ ] Review the pull request.

Closes #9.

### CODE REVIEW

File: saber/tests/test_sequence_processor.py [hunk 1]
```diff
@@ -103,12 +103,11 @@ def test_attributes_after_initilization_of_model(sp_no_ds_no_embed, dummy_config
     assert sp_no_ds_no_embed.config.grad_norm == 1.0
     assert sp_no_ds_no_embed.config.k_folds == 2
     assert sp_no_ds_no_embed.config.learning_rate == 0.0
-    assert not sp_no_ds_no_embed.config.load_pretrained_model
     assert sp_no_ds_no_embed.config.epochs == 50
     assert sp_no_ds_no_embed.config.model_name == 'mt-lstm-crf'
     assert sp_no_ds_no_embed.config.optimizer == 'nadam'
     assert sp_no_ds_no_embed.config.output_folder == abspath('../output')
-    assert sp_no_ds_no_embed.config.pretrained_model_weights is None
+    assert sp_no_ds_no_embed.config.pretrained_model_weights is ''
```
[the remark this answers is not in the document]
Comment (JohnGiorgi) [reply saber/tests/test_sequence_processor.py]: `assert` used in `pytest` framework is normal, ignore.

File: saber/tests/test_multi_task_lstm_crf.py [hunk 2]
```diff
@@ -56,12 +56,11 @@ def test_model_attributes_after_creation_of_model(multi_task_lstm_crf_single_mod
     assert multi_task_lstm_crf_single_model.config.grad_norm == 1.0
     assert multi_task_lstm_crf_single_model.config.k_folds == 2
     assert multi_task_lstm_crf_single_model.config.learning_rate == 0.0
-    assert not multi_task_lstm_crf_single_model.config.load_pretrained_model
     assert multi_task_lstm_crf_single_model.config.epochs == 50
     assert multi_task_lstm_crf_single_model.config.model_name == 'mt-lstm-crf'
     assert multi_task_lstm_crf_single_model.config.optimizer == 'nadam'
     assert multi_task_lstm_crf_single_model.config.output_folder == abspath('../output')
-    assert multi_task_lstm_crf_single_model.config.pretrained_model_weights is None
+    assert multi_task_lstm_crf_single_model.config.pretrained_model_weights == ''
```
[the remark this answers is not in the document]
Comment (JohnGiorgi) [reply saber/tests/test_multi_task_lstm_crf.py]: `assert` used in `pytest` framework is normal, ignore.
